staging/src/k8s.io/apiserver/pkg/endpoints/handlers/rest.go

```diff
@@ -457,6 +458,65 @@ func isDryRun(url *url.URL) bool {
 	return len(url.Query()["dryRun"]) != 0
 }
 
+// fieldValidation checks if the fieldValidation query parameter is set on the request,
+// and if so ensures that the parameter is valid and that the request has a valid
+// media type, because the list of media types that support field validation are a subset of
+// all supported media types (protobuf does not support field validation).
+func fieldValidation(req *http.Request) (runtime.FieldValidationDirective, error) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.FieldValidation) {
+		return runtime.IgnoreFieldValidation, nil
+	}
+
+	// TODO: Should we blocklist unsupportedContentTypes (just protobuf) rather than allowlisting everything that isn't protobuf?
```

Comment:
Since this is not a feature that we'll typically get automatically for each new content type (we need to make sure that we can run a strict version of the decoder), I would do an allowlist rather than a blacklist.